invenio_records_resources/services/custom_fields/mappings.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 CERN.
+#
+# Invenio-Records-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Custom fields mappings."""
+
+from abc import abstractmethod
+
+
+class Mapping:
+    """Base class for mappings."""
+
+    @abstractmethod
+    def mapping(self):
```

Comment:
```suggestion
    def to_dict(self):
```

got weirded out at some point :)